apiserver/pgadminservice/pgadminimpl.go

```diff
@@ -0,0 +1,285 @@
+package pgadminservice
+
+/*
+Copyright 2018 - 2020 Crunchy Data Solutions, Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import (
+	"fmt"
+
+	crv1 "github.com/crunchydata/postgres-operator/apis/crunchydata.com/v1"
+	"github.com/crunchydata/postgres-operator/apiserver"
+	msgs "github.com/crunchydata/postgres-operator/apiservermsgs"
+	"github.com/crunchydata/postgres-operator/config"
+	"github.com/crunchydata/postgres-operator/internal/pgadmin"
+	"github.com/crunchydata/postgres-operator/kubeapi"
+
+	log "github.com/sirupsen/logrus"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const pgAdminServiceSuffix = "-pgadmin"
+
+// CreatePgAdmin ...
+// pgo create pgadmin mycluster
+// pgo create pgadmin --selector=name=mycluster
+func CreatePgAdmin(request *msgs.CreatePgAdminRequest, ns, pgouser string) msgs.CreatePgAdminResponse {
+	var err error
+	resp := msgs.CreatePgAdminResponse{
+		Status:  msgs.Status{Code: msgs.Ok},
+		Results: []string{},
+	}
+
+	log.Debugf("createPgAdmin selector is [%s]", request.Selector)
+
+	// try to get the list of clusters. if there is an error, put it into the
+	// status and return
+	clusterList, err := getClusterList(request.Namespace, request.Args, request.Selector)
+	if err != nil {
+		resp.SetError(err.Error())
+		return resp
+	}
+
+	for _, cluster := range clusterList.Items {
+		log.Debugf("adding pgAdmin to cluster [%s]", cluster.Name)
```

Comment:
A small check needs to be added to keep createPgAdmin from running against a cluster from an earlier Operator release that has yet to be upgraded. It should be similar to this (pulled from `apiserver/failoverservice/failoverimple.go`:
```
        // check if the current cluster is not upgraded to the deployed
	// Operator version. If not, do not allow the command to complete
	if cluster.Annotations[config.ANNOTATION_IS_UPGRADED] == config.ANNOTATIONS_FALSE {
		resp.Status.Code = msgs.Error
		resp.Status.Msg = cluster.Name + msgs.UpgradeError
		return response
	}
```